### PR TITLE
bootstrap-tools-cross: Rename bootstrap file jobs to be more descriptive

### PR DESCRIPTION
**Pull request by Artturin**
Rename files in `bootstrap-files/` to match jobs in `make-bootstrap-tools-cross`.

Should make automating the bootstrap files easier.

```nix
$ nix repl --file ./pkgs/top-level/release-cross.nix
nix-repl> bootstrapTools
{ aarch64-linux = { ... }; aarch64-linux-musl = { ... }; armv5tel-linux = { ... }; armv6l-linux = { ... }; armv6l-linux-musl = { ... }; armv7l-linux = {... }; mips64el-linux-gnuabi64 = { ... }; mips64el-linux-gnuabin32 = { ... }; mipsel-linux-gnu = { ... }; powerpc64le-linux = { ... }; riscv64-linux = {... }; x86_64-linux-musl = { ... }; }
```

Additionally if non-linux bootstrap files are added to `make-bootstrap-tools-cross` then there won't be any renaming needed.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/stdenv/linux/make-bootstrap-tools-cross.nix [hunk 1]
```diff
@@ -15,18 +15,18 @@ in lib.mapAttrs (n: make) (with lib.systems.examples; {
...
-  armv5tel   = sheevaplug;
-  armv6l     = raspberryPi;
-  armv7l     = armv7l-hf-multiplatform;
-  aarch64    = aarch64-multiplatform;
-  mipsel-linux-gnu         = mipsel-linux-gnu;
+  armv5tel-linux   = sheevaplug;
+  armv6l-linux     = raspberryPi;
+  armv7l-linux     = armv7l-hf-multiplatform;
+  aarch64-linux    = aarch64-multiplatform;
+  mipsel-linux        = mipsel-linux-gnu;
```
Comment (Artturin) [inline pkgs/stdenv/linux/make-bootstrap-tools-cross.nix:22]: Removed the `-gnu` because it does not seem significant

**Comment by alyssais** [comment]
@OfBorg eval

**Comment by alyssais** [comment]
> IMO `-gnu` is not added because that will likely be the default in nixpkgs always and other chains should do it like musl.

I would much prefer we always use a full triple, e.g. `x86_64-unknown-linux-gnu`.  If you want easier automation, it'll be easier if every bootstrap platform consistently uses the same format.

**Review by alyssais — APPROVED** [review]
LGTM.  Maybe needs a squash?

**Review by Ericson2314 — APPROVED** [review]

File: pkgs/stdenv/linux/default.nix [hunk 2]
```diff
@@ -59,24 +59,24 @@
...
+      armv7l-linux = import ./bootstrap-files/armv7l-unknown-linux-gnueabihf.nix;
+      aarch64-linux = import ./bootstrap-files/aarch64-unknown-linux-gnu.nix;
+      mipsel-linux = import ./bootstrap-files/mipsel-unknown-linux-gnu.nix;
       mips64el-linux = import
        (if localSystem.isMips64n32
-        then ./bootstrap-files/mips64el-n32.nix
-        else ./bootstrap-files/mips64el.nix);
-      powerpc64le-linux = import ./bootstrap-files/powerpc64le.nix;
-      riscv64-linux = import ./bootstrap-files/riscv64.nix;
+        then ./bootstrap-files/mips64el-unknown-linux-gnuabin32.nix.nix
```
Comment (amjoseph-nixpkgs) [inline pkgs/stdenv/linux/default.nix:71]:
```suggestion
        then ./bootstrap-files/mips64el-unknown-linux-gnuabin32.nix
```

Comment (Artturin) [reply pkgs/stdenv/linux/default.nix:71]: https://github.com/NixOS/nixpkgs/pull/262382

**Pull request merged**